esim-cloud-backend/arduinoAPI/tasks.py

```diff
@@ -0,0 +1,110 @@
+import os
+import shutil
+from pathlib import Path
+import time
+import subprocess
+import traceback
+from django.conf import settings
+from celery import shared_task, current_task
+from celery import states
+from celery.exceptions import Ignore
+import json
+import logging
+import re
+
+logger = logging.getLogger(__name__)
+
+
+def saveFiles(data):
+    try:
+        filenames = []
+        if not os.path.exists(settings.MEDIA_ROOT):
+            Path(settings.MEDIA_ROOT).mkdir(parents=True, exist_ok=True)
+        for k in data:
+            work_dir = settings.MEDIA_ROOT+'/'+str(k)
+
+            Path(work_dir).mkdir(parents=True, exist_ok=True)
+
+            filename = settings.MEDIA_ROOT+'/'+str(k)+'/sketch.ino'
+            fout = open(filename, 'w', encoding='utf8')
+            fout.writelines('#line 1 "{}"\n'.format(filename))
```

Comment:
any reason not to use the updated file handling method of 
```
with open():
     steps
```

Reply:
no specific reason. I just prefer the old one.